test/common/verify_alert_links_to_SOPs.go

```diff
@@ -0,0 +1,147 @@
+package common
+
+import (
+	"encoding/json"
+	"fmt"
+	prometheusv1 "github.com/prometheus/client_golang/api/prometheus/v1"
+	"io"
+	"net/http"
+	"os"
+	"strings"
+)
+
+var githubToken = os.Getenv("TOKEN_GITHUB")
+
+func TestVerifyAlertLinksInSOPs(t TestingTB, ctx *TestingContext) {
+
+	if githubToken == "" {
+		t.Skip("github token not present")
+	}
+
+	var apiLinks []string
+	testUrl := "https://github.com/RHCloudServices/integreatly-help/blob/master/sops/README.md"
+	testResp := getSOPAlertLinkStatus(t, testUrl)
+
+	if testResp.StatusCode != 200 {
+		t.Log("Response status: ", testUrl, testResp.Status)
+	} else {
+		t.Log("Response status: ", testResp.Status)
+	}
+
+	output, err := execToPod("wget -qO - localhost:9090/api/v1/rules",
+		"prometheus-prometheus-0",
+		ObservabilityProductNamespace,
+		"prometheus", ctx)
+	if err != nil {
+		t.Fatal("failed to exec to pod:", err)
+	}
+
+	var ApiOutput prometheusAPIResponse
+
+	err = json.Unmarshal([]byte(output), &ApiOutput)
+	if err != nil {
+		t.Fatal("Failed to unmarshal json", err)
+	}
+
+	var rulesResult prometheusv1.RulesResult
+
+	err = json.Unmarshal([]byte(ApiOutput.Data), &rulesResult)
+	if err != nil {
+		t.Fatal("Failed to unmarshal json", err)
+	}
+
+	for _, group := range rulesResult.Groups {
+		for _, rule := range group.Rules {
+			switch v := rule.(type) {
+			case prometheusv1.RecordingRule:
+			case prometheusv1.AlertingRule:
+				for annotation, sopUrl := range v.Annotations {
+					if annotation == "sop_url" {
+						apiLinks = append(apiLinks, string(sopUrl))
+					}
+
+				}
+
+			default:
+				fmt.Printf("Unknown rule type %s", v)
+
+			}
+		}
+	}
+
+	apiLinks = unique(apiLinks)
+	result := validateSOPLinks(t, apiLinks)
+	fmt.Println(result)
+}
+
+func modifyLink(rawLink string) (url string) {
+	r := strings.NewReplacer(
+		"github", "api.github",
+		"com/", "com/repos/",
+		"blob/master", "contents",
+		"tree/master", "contents",
+	)
+
+	url = r.Replace(rawLink)
+	return url
```

Comment:
I think you need just `return` here. The reason is that you specified that you want to return `url` in the function declaration (line 77). Alternatively just use `func modifyLink(rawLink string) string {` and keep the `return url` as is. See e.g. https://www.geeksforgeeks.org/named-return-parameters-in-golang/?ref=lbp